src/trackinfoobject.cpp

```diff
@@ -722,20 +722,28 @@ QString TrackInfoObject::getURL() {
 }
 
 ConstWaveformPointer TrackInfoObject::getWaveform() {
+    QMutexLocker lock(&m_qMutex);
```

Comment:
I'm not sure. Almost all the other geters and setters use locks, so I added one here also.